ocpp-v1_6/src/main/java/eu/chargetime/ocpp/model/core/StopTransactionRequest.java

```diff
@@ -72,10 +72,10 @@ public IdToken getIdTag() {
      * It is optional because a Charge Point may terminate charging without the presence of an idTag, e.g. in case of a reset.
      * A Charge Point SHALL send the idTag if known.
      *
-     * @param idTag the {@link IdToken}.
+     * @param idTag the IdToken.
      */
     @XmlElement
-    public void setIdTag(IdToken idTag) {
+    public void setIdTag(String idTag) {
```

Comment:
This method should validate the input and throw an exception if it's invalid. Fail fast. This should be verified by test.
Also, is this check in the in validate method?